hazelcast-sql-core/src/test/java/com/hazelcast/sql/impl/expression/string/PositionFunctionIntegrationTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.string;
+
+import com.hazelcast.sql.HazelcastSqlException;
+import com.hazelcast.sql.SqlColumnType;
+import com.hazelcast.sql.SqlRow;
+import com.hazelcast.sql.impl.SqlDataSerializerHook;
+import com.hazelcast.sql.impl.expression.ConstantExpression;
+import com.hazelcast.sql.impl.expression.ExpressionTestSupport;
+import com.hazelcast.sql.impl.type.QueryDataType;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class PositionFunctionIntegrationTest extends ExpressionTestSupport {
+
+    @Test
+    public void test() {
+        put("XYZ");
+
+        check(sql("'ABCD'", "'BC'"), 2);
+        check(sql("'ABCD'", "''"), 1);
+        check(sql("'ABCD'", "'NO_MATCH'"), 0);
+    }
+
+    @Test
+    public void test_null() {
+        put("XYZ");
+
+        check(sql("'BC'", "NULL"), null);
+        check(sql("NULL", "'BC'"), null);
+        check(sql("NULL", "NULL"), null);
+    }
+
+    @Test
+    public void test_parameter() {
+        put("XYZ");
+
+        check(sql("?", "'BC'"), 2, "ABCD");
+        check(sql("'ABCD'", "?"), 2, "BC");
+        check(sql("?", "?"), 2, "BC", "ABCD");
+    }
+
+    @Test
+    public void test_wrong_type() {
+        put("XYZ");
+
+        checkError(sql("123", "23"));
+        checkError(sql("TRUE", "FALSE"));
+        checkError(sql("?", "?"), 123, 23);
```

Comment:
Just asking for myself : did you test arguments with different types in your POSITION function?